lib/galaxy/tool_shed/galaxy_install/tool_dependencies/recipe/install_environment.py

```diff
@@ -246,18 +248,18 @@ def handle_complex_command(self, command, job_name=""):
 
     def log_results(self, command, fabric_AttributeString, file_path):
         """Write attributes of fabric.operations._AttributeString to a specified log file."""
-        mode = 'ab' if os.path.exists(file_path) else 'wb'
+        mode = 'a' if os.path.exists(file_path) else 'w'
```

Comment:
My thought here was that it is better to keep this as bytes, given that this is used when compiling packages, which may emit non-unicode errors. At least we'd have the unaltered log file on disk then. On the other hand we don't really use this anymore, so ... 🤷